app/(auth)/index.tsx

```diff
@@ -0,0 +1,209 @@
+  // const [password, onChangePassword] = useState('');
+  const [buttonDisabled, setButtonDisabled] = useState(true);
+  const { signIn } = useSession();
+
+  function checkEmail(email: string) {
+    let re =
+      /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+    if (re.test(email)) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+  const dispatch: Dispatch<any> = useDispatch();
+
+  function auth() {
+    // signIn({ email: email, password: password });
+    dispatch(reduxSignIn(email));
+  }
+
+  useEffect(() => {
+    if (password !== '' && checkEmail(email)) {
+      setButtonDisabled(false);
+    } else {
+      setButtonDisabled(true);
+    }
+  }, [email, password]);
+  const ref_input2 = useRef<TextInput | null>(null);
+  return (
+    <View style={{ flex: 1, display: 'flex', gap: 60, backgroundColor: '#7b7776' }}>
+      <KeyboardAvoidingView
+        style={styles.container}
+        behavior={Platform.OS === 'ios' ? 'padding' : 'height'}
+      >
+        <View style={styles.mainComponent}>
+          <View style={styles.mainScreen}>
+            <LED />
+            <View style={styles.titleContainer}>
+              <Text style={{ fontFamily: 'retroGaming' }}>Connectez vous</Text>
+            </View>
+            <View style={styles.inputView}>
+              <Text style={styles.placeholder}>Identifiant</Text>
+              <TextInput
+                testID="inputEmail"
+                value={email}
+                style={styles.input}
+                returnKeyType="next"
+                onSubmitEditing={() => ref_input2.current?.focus()}
+                blurOnSubmit={false}
+                onChangeText={onChangeEmail}
+              />
+            </View>
+            <View style={styles.inputView}>
+              <Text style={styles.placeholder}>Mot de passe</Text>
+              <TextInput
+                testID="inputPassword"
+                ref={ref_input2}
+                value={password}
+                style={styles.input}
+                secureTextEntry={true}
+                onChangeText={onChangePassword}
+                onSubmitEditing={() => {
+                  auth();
+                }}
+              />
+            </View>
+          </View>
+
+          {/* <TouchableOpacity
+            testID="submitButton"
+            disabled={buttonDisabled}
+            style={[
+              styles.button,
+              buttonDisabled ? { backgroundColor: '#cf95b2ff' } : { backgroundColor: '#DB1778' },
+            ]}
+            onPress={() => {
+              signIn({ email: email, password: password });
+            }}
+          >
+            <Text style={{ color: 'white' }}>Se connecter</Text>
+          </TouchableOpacity> */}
+        </View>
+      </KeyboardAvoidingView>
+      <PokedexButtom buttonDisabled={buttonDisabled} signIn={() => auth()} />
```

Comment:
Evite de faire ça, ça recréer une fonction à chaque rendu et ça empèche l'optimisation que tu pourrais mettre en place dans auth().

```suggestion
      <PokedexButtom buttonDisabled={buttonDisabled} signIn={auth)} />
```